framework/src/application/network/network.ts

```diff
@@ -308,13 +313,25 @@ export class Network {
 			if (request.wasResponseSent) {
 				return;
 			}
-			// eslint-disable-next-line @typescript-eslint/prefer-includes
-			const hasTargetModule = hasNamespaceReg.test(request.procedure);
-			// If the request has no target module, default to app (to support legacy protocol).
-			const sanitizedProcedure = hasTargetModule ? request.procedure : `app:${request.procedure}`;
+
+			if (!REMOTE_ACTIONS_WHITE_LIST.includes(request.procedure)) {
+				const error = new Error(`Requested procedure "${request.procedure}" is not permitted.`);
+				this._logger.error(
+					{ err: error, procedure: request.procedure },
+					'Peer request not fulfilled event: Requested procedure is not permitted.',
+				);
+
+				// Ban peer on if non-permitted procedure is requested
+				this._p2p.applyPenalty({ peerId: request.peerId, penalty: 100 });
+
+				// Send an error back to the peer.
+				request.error(error);
+				return;
+			}
+
 			try {
-				const result = await this._channel.invokePublic<liskP2P.p2pTypes.P2PNodeInfo>(
-					sanitizedProcedure,
+				const result = await this._channel.invoke<liskP2P.p2pTypes.P2PNodeInfo>(
+					`app:${request.procedure}`,
```

Comment:
we can also remove `app:` prefix from here?

Reply:
No its necessary here to route right action to the Bus. As of now `Bus` and `Channel` are all have a particular prefix, if we remove it will throw error to not able to identify action.  @ishantiw @shuse2